design/ceph/admission-controller.md

```diff
@@ -0,0 +1,93 @@
+#Admission Controllers on Rook
+
+#Background
+This proposal is to add support for admission controllers in Rook. 
+
+Currently, user can manipulate CRD specs with any values which may result in application not functioning as expected. An admission controller is a piece of code that intercepts requests to the Kubernetes API server prior to persistence of the object, but after the request is authenticated and authorized. In that list, there are two special controllers: MutatingAdmissionWebhook and ValidatingAdmissionWebhook. These execute the mutating and validating (respectively) admission control webhooks which are configured in the API.
+
+The admission control process proceeds in two phases. In the first phase, mutating admission controllers are run. In the second phase, validating admission controllers are run. Note again that some of the controllers are both.
+
+If any of the controllers in either phase reject the request, the entire request is rejected immediately and an error is returned to the end-user.
+
+# Requirements
+
+1. Webhook server which will validate the requests
+2. TLS certificates for the server,
+3. ValidatingWebhookConfig/MutatingWebhookConfig which will intercept requests and send a HTTPS request to the webhook server.
+4. RBAC Components 
+
+#Admission Controllers in Rook
+
+![AC Image](./design-pic.png)
+
+##Setup
+###Certificate Management
+####Development and Testing 
+![AC Image](./ac_dev_design.png)
+
+We will be opting for using for self-signed certificates approved by the kubernetes cluster itself. Once common.yaml and the operator has been deployed.  Using scripts we will be performing the following tasks
+* Creating the private key and certs using openssl with the names tls.key and tls.crt. We will be hardcoding these names for now as it will be explained in later section
+* Creating and sending a CSR to kubernetes for approval
+* Once approved, a generic secret will be created with the given tls.key and tls.crt which will be later mounted onto the server pod for use.
+* Modifying the webhook.yaml to inject CA bundle onto the ValidatingWebhookConfig
+* Creating a webhook deployment, service, config on the cluster and deploying the entire webhook setup
+* Now our webhook server has been configured, deployed and ready to intercept requests based on the webhookconfig.
+* **All the above resources will be created in rook-ceph namespace hence it should be created in prior**
```

Comment:
> it should be created in prior

What do you mean? The rook-ceph namespace just already needs to be created first?

Reply:
Yes travis, as all the components will be created in rook-ceph namespace. The namespace would need to have been already created but since the operator would be creating the admission controller, i guess the the namespace would already exist. As we are doing this from this scripts for dev/testing purpose, i thought it be best to add that point.  I'll be modifying it !

Reply:
Fixed